builtin/logical/pki/path_issue_sign.go

```diff
@@ -0,0 +1,208 @@
+defaults to "pem".`,
+	},
+}
+
+func pathIssue(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "issue/" + framework.GenericNameRegex("role"),
+		Fields:  issueAndSignSchema,
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.WriteOperation: b.pathIssue,
+		},
+
+		HelpSynopsis:    pathIssueHelpSyn,
+		HelpDescription: pathIssueHelpDesc,
+	}
+}
+
+func pathSign(b *backend) *framework.Path {
+	ret := &framework.Path{
+		Pattern: "sign/" + framework.GenericNameRegex("role"),
+		Fields:  issueAndSignSchema,
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.WriteOperation: b.pathSign,
+		},
+
+		HelpSynopsis:    pathSignHelpSyn,
+		HelpDescription: pathSignHelpDesc,
+	}
+
+	ret.Fields["csr"] = &framework.FieldSchema{
+		Type:        framework.TypeString,
+		Default:     "",
+		Description: `PEM-format CSR to be signed.`,
+	}
+
+	return ret
+}
+
+func (b *backend) pathIssue(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	return b.pathIssueSignCert(req, data, false)
+}
+
+func (b *backend) pathSign(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	return b.pathIssueSignCert(req, data, true)
+}
+
+func (b *backend) pathIssueSignCert(
+	req *logical.Request, data *framework.FieldData, useCSR bool) (*logical.Response, error) {
+	roleName := data.Get("role").(string)
+
+	// Get the role
+	role, err := b.getRole(req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("Unknown role: %s", roleName)), nil
+	}
+
+	format := data.Get("format").(string)
+	switch format {
+	case "pem":
+	case "der":
+	default:
+		return logical.ErrorResponse(
+			`The "format" path parameter must be "pem" or "der"`), nil
+	}
+
+	var caErr error
+	signingBundle, caErr := fetchCAInfo(req)
+	switch caErr.(type) {
+	case certutil.UserError:
+		return nil, certutil.UserError{Err: fmt.Sprintf(
+			"Could not fetch the CA certificate (was one set?): %s", caErr)}
+	case certutil.InternalError:
+		return nil, certutil.InternalError{Err: fmt.Sprintf(
+			"Error fetching CA certificate: %s", caErr)}
+	}
+
+	var parsedBundle *certutil.ParsedCertBundle
+	if useCSR {
+		parsedBundle, err = signCert(b, role, signingBundle, false, false, req, data)
+	} else {
+		parsedBundle, err = generateCert(b, role, signingBundle, false, req, data)
+	}
+	if err != nil {
+		switch err.(type) {
+		case certutil.UserError:
+			return logical.ErrorResponse(err.Error()), nil
+		case certutil.InternalError:
+			return nil, err
+		}
+	}
+
+	cb, err := parsedBundle.ToCertBundle()
+	if err != nil {
+		return nil, fmt.Errorf("Error converting raw cert bundle to cert bundle: %s", err)
+	}
+
+	resp := b.Secret(SecretCertsType).Response(
+		structs.New(cb).Map(),
+		map[string]interface{}{
+			"serial_number": cb.SerialNumber,
+		})
+
+	if format == "der" {
```

Comment:
No, PEM is the format that the values are already in, and the default option for the format field.